ScriptingExample/src/test/java/org/openzen/zenscript/scriptingexample/tests/actual_test/arrays/ArrayOperatorTests.java

```diff
@@ -0,0 +1,123 @@
+		logger.assertPrintOutput(0, "true");
+	}
+
+	@Test
+	public void containsReturnsFalseForNonMatch() {
+		ScriptBuilder.create()
+				.add("var array = ['a', 'b', 'c'] as string[];")
+				.add("println(('d' in array) as string);")
+				.execute(this);
+
+		logger.assertPrintOutputSize(1);
+		logger.assertPrintOutput(0, "false");
+	}
+
+	@Test
+	public void canCastToList() {
+		ScriptBuilder.create()
+				.add("var array = ['a', 'b', 'c'] as string[];")
+				.add("var list = array as stdlib.List<string>;")
+				.add("var listImplicit as stdlib.List<string>;")
+				.add("listImplicit = array;")
+				.add("println(list.length);")
+				.add("println(listImplicit.length);")
+				.execute(this);
+
+		logger.assertPrintOutputSize(2);
+		logger.assertPrintOutput(0, "3");
+		logger.assertPrintOutput(1, "3");
+	}
+
+	@Test
+	public void canCastFromList() {
+		ScriptBuilder.create()
+				.add("var list = new stdlib.List<string>();")
+				.add("list.add('a');")
+				.add("list.add('b');")
+				.add("list.add('c');")
+				.add("var array = list as string[];")
+				.add("var arrayImplicit as string[];")
+				.add("arrayImplicit = list;")
+				.add("println(array.length);")
+				.add("println(arrayImplicit.length);")
+				.execute(this);
+
+		logger.assertPrintOutputSize(2);
+		logger.assertPrintOutput(0, "3");
+		logger.assertPrintOutput(1, "3");
+	}
+
+	@Test
+	public void indexMethodWorks() {
+		ScriptBuilder.create()
+				.add("var array = ['A', 'B', 'C'] as string[];")
+				.add("var indexer as function(element as string) as string = (element as string) as string => element;")
+				.add("var map = array.index<string>(indexer);")
+				.add("for key, value in map println (key + ': ' + value);")
+				.execute(this);
+
+		logger.assertPrintOutputSize(3);
+		logger.assertPrintOutput(0, "A: A");
+		logger.assertPrintOutput(1, "B: B");
+		logger.assertPrintOutput(2, "C: C");
+	}
+
+	@Test
+	public void mapWorks() {
+		ScriptBuilder.create()
+				.add("var array = ['A', 'B', 'C'] as string[];")
+				.add("var mapper as function(element as string) as string = (element as string) as string => element + element;")
+				.add("var mapped = array.map<string>(mapper);")
+				.add("for element in mapped println(element);")
+				.execute(this);
+
+		logger.assertPrintOutputSize(3);
+		logger.assertPrintOutput(0, "AA");
+		logger.assertPrintOutput(1, "BB");
+		logger.assertPrintOutput(2, "CC");
+	}
+
+	@Test
+	public void mapWithIndexWorks() {
+		ScriptBuilder.create()
+				.add("var array = ['A', 'B', 'C'] as string[];")
+				.add("var mapper as function(index as usize, element as string) as string = (index as usize, element as string) as string => element + index + element;")
+				.add("var mapped = array.map<string>(mapper);")
+				.add("for element in mapped println(element);")
+				.execute(this);
+
+		logger.assertPrintOutputSize(3);
+		logger.assertPrintOutput(0, "A0A");
+		logger.assertPrintOutput(1, "B1B");
+		logger.assertPrintOutput(2, "C2C");
+	}
```

Comment:
You could add tests for the other means of creating arrays (both single, and multi-dimensional):
- [x] Literal: `var x = [1, 2, 3]`
- [ ] Constructor with default value
     - [ ] 1D: `var x = new int[](size, defaultValue); println(x[0]);`
     - [ ] 2D: `var x = new int[,](sizeA, sizeB, defaultValue); println(x[0,0]);
- [ ] Constructor with callback function `var x = new int[](size, (index as usize) => {return 10 as int;}); println(x[0]);`
- [ ] Constructor with projection function `var x = [1, 2, 3] as int[]; var y = new int[](x, (valueFromX as int) => {return 2 * valueFromX;}); println(y[0]);`